etl/nixpkgs-graph.nix

```diff
@@ -1,17 +1,38 @@
 # usage:
-# NIXPKGS_FLAKE_REF="github:nixos/nixpkgs/master" nix eval --json --file "./nixpkgs-graph.nix"
+# NIXPKGS_ALLOW_BROKEN=1 NIXPKGS_ALLOW_INSECURE=1 TARGET_FLAKE_REF="github:nixos/nixpkgs/master" TARGET_FLAKE_SYSTEM="x86_64-linux" nix eval --json --file "./nixpkgs-graph.nix"
 
 let
-  nixpkgsFlakeRef = builtins.getEnv "NIXPKGS_FLAKE_REF";
-  pkgs = import (builtins.getFlake nixpkgsFlakeRef) {
-    config = {
-      # package 'python-2.7.18.6' is marked as insecure
-      # we need to allow this package otherwise it is refusing to evaluate
-      permittedInsecurePackages = [
-        "python-2.7.18.6"
-      ];
-    };
-  };
+  pkgs = builtins.getFlake "nixpkgs";
+  targetFlakeRef = builtins.getEnv "TARGET_FLAKE_REF";
+  # We specify the system of the target flake here to avoid ambiguity
+  # An output will produce a different result on every system
+  targetFlakeSys = builtins.getEnv "TARGET_FLAKE_SYSTEM";
+  targetFlakePkgs =
+    if (builtins.getFlake targetFlakeRef).outputs ? packages
+    then
+      if (builtins.getFlake targetFlakeRef).outputs.packages ? ${targetFlakeSys}
+      then
+        (builtins.getFlake targetFlakeRef).outputs.packages.${targetFlakeSys}
+      else
+        { }
```

Comment:
I'm sure there is a way to do it in one go instead of many ifs, cc @thufschmitt

Reply:
I think you can do `(builtins.getFlake targetFlakeRef).outputs.packages.${targetFlakeSys} or {}`